pkg/util/kernel/headers.go

```diff
@@ -0,0 +1,157 @@
+// +build linux
+
+package kernel
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host"
+	"github.com/mholt/archiver/v3"
+)
+
+const sysfsHeadersPath = "/sys/kernel/kheaders.tar.xz"
+const kernelModulesPath = "/lib/modules/%s/build"
+
+func FindHeaderDirs() ([]string, error) {
+	hv, err := HostVersion()
+	if err != nil {
+		return nil, fmt.Errorf("unable to determine host kernel version: %w", err)
+	}
+
+	dir, err := getHeaderDirs(hv)
+	if err == nil {
+		return dir, nil
+	}
+
+	if os.IsNotExist(err) {
+		// as last resort, look for sysfs headers
+		if dir, syserr := getSysfsHeaderDirs(hv); syserr == nil {
+			return dir, nil
+		}
+	}
+	return nil, err
+}
+
+func getHeaderDirs(v Version) ([]string, error) {
+	hi := host.GetStatusInformation()
+	if hi.KernelVersion == "" {
+		return nil, fmt.Errorf("unable to get host metadata")
+	}
+
+	dirs := []string{kernelModulesPath}
```

Comment:
I was a bit thrown by how this was written, could we do something like:

```
dirs := []string{fmt.Sprintf(kernelModulesPath, hi.KernelVersion}}
if hi.Platform == "debian" {
   dirs = append(dirs, fmt.Sprintf("/lib/modules/%s/source", hi.KernelVersion))
}
```

Reply:
Sure.